### PR TITLE
remove m2o from supportedTypes in selection widget

### PR DESCRIPTION
**Pull request by msh-odoo**
Task:https://www.odoo.com/web#id=1871698&action=333&active_id=1724&model=project.task&view_type=form&menu_id=4720

Pad: https://pad.odoo.com/p/r.e9a827ef3eb6c5997c6fe672f7181417

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: doc/reference/javascript_reference.rst [hunk 1]
```diff
@@ -1697,7 +1697,7 @@ Relational fields
 - many2one (FieldMany2One)
     Default widget for many2one fields.
 
-    - Supported field types: *selection, many2one*
+    - Supported field types: *many2one*
```
Comment (aab-odoo) [inline doc/reference/javascript_reference.rst]: also remove `many2one` line 1669

Comment (msh-odoo) [reply doc/reference/javascript_reference.rst]: @aab-odoo Done

**Comment by aab-odoo** [comment]
robodoo r+

**Comment by pedrobaeza** [comment]
But isn't payment term on res.partner using that widget?

**Comment by msh-odoo** [comment]
@pedrobaeza this will not stop you to use it from xml, this will not allow user to add this widget from studio

**Comment by pedrobaeza** [comment]
Ah, OK, thanks for the clarification. Any way, as extended comment, that widget is in general conflicting, as only show first N results, and I have customers with hundreds of payment terms (due to several payment days), and they are not able to use the system. I needed to change the widget to regular many2one in these cases.

**Pull request closed without merging**